src/tests/ports/trendings.spec.ts

```diff
@@ -115,86 +152,264 @@ test('trendings component', function ({ components }) {
+                getItemWithPickStats(
+                  sale.searchContractAddress,
+                  sale.searchItemId!
+                )
+              )
+          })
+
+          it('should return the trending results by sales and volume shuffled', async () => {
+            // 40% of the big volumes items should be in the trendings array
+            expect(trendings).toEqual(
+              expect.arrayContaining(trendingItemsWithMostVolume)
+            )
+
+            // the rest of the big volume sales shouldn't be in the trendings array
+            expect(trendings).not.toEqual(
+              expect.arrayContaining(notTrendingItemsWithMostVolume)
+            )
+
+            // the trendings but not in sale, should not be part of the results
+            expect(trendings).not.toEqual(
+              expect.arrayContaining(trendingItemButNotInSale)
+            )
+
+            // 60% of the big amount of sales items should be in the trendings array
+            expect(trendings).toEqual(
+              expect.arrayContaining(trendingItemsWithMostSales)
+            )
+            // the rest of the big amount of sales items shouldn't be in the trendings array
+            expect(trendings).not.toEqual(
+              expect.arrayContaining(notTrendingItemsWithMostSales)
+            )
+          })
+
+          it('should fetch the data from the subgraph', async () => {
+            // assert the retry calls, it should ask for all pages
+            const { collectionsSubgraph } = components
+
+            expect(collectionsSubgraph.query).toHaveBeenCalledTimes(
+              Math.ceil(salesResponse.length / 1000)
+            )
+
+            Array.from(
+              { length: Math.ceil(salesResponse.length / 1000) }, // the mock is returning 10000 results
+              (_, i) => {
+                expect(collectionsSubgraph.query).toHaveBeenCalledWith(
+                  getTrendingsQuery({
+                    from: getDateXDaysAgo(1).getTime(),
+                    first: 1000,
+                    skip: 1000 * i, // 1000 is the max number of results per page
+                  })
+                )
+              }
+            )
+          })
+
+          it('should fetch the data from favorites component', async () => {
+            const { favorites } = components
+            expect(favorites.getPicksStatsOfItems).toHaveBeenCalledWith(
+              itemIds,
+              undefined
+            )
+          })
+        })
+
+        describe('and there is a filter of "pickedBy"', () => {
+          let pickedBy: string
+
+          beforeEach(async () => {
+            pickedBy = '0x123'
+            filters = { ...filters, pickedBy }
+            trendings = await trendingsComponent.fetch(filters)
+          })
+
+          it('should fetch the data from favorites component', async () => {
+            const { favorites } = components
+            expect(favorites.getPicksStatsOfItems).toHaveBeenCalledWith(
```

Comment:
I think that we could improve this test by checking if the trendings have or not the picks property with their correspondent picks. This is important because although we can call the API, what we want to be sure about is that the component returns the picks. What do you think about adding this test?

Reply:
I think you're right. I've added a new test checking if the trending items have their correspondent picks stats.
Let me know wdyt 🙏